process/smartContract/process.go

```diff
@@ -1434,11 +1505,9 @@ func (sc *scProcessor) createSmartContractResults(
 			result.OriginalSender = tx.GetSndAddr()
 		}
 
-		isAsyncTransferBackToSender := callType == vmcommon.AsynchronousCall &&
-			bytes.Equal(outAcc.Address, tx.GetSndAddr())
 		isLastOutTransfer := i == lenOutTransfers-1
-		if isLastOutTransfer && isAsyncTransferBackToSender && sc.isTransferWithNoDataOrBuiltInCall(outputTransfer.Data) {
-			addVMOutputResultsToSCR(vmOutput, result)
```

Comment:
Not need to call addVMOutputResultsToSCR? In line 1548 and 1549 it is done now only part of this whole method. Also 
is done result.GasLimit += vmOutput.GasRemaining instead result.GasLimit = vmOutput.GasRemaining like it is done in addVMOutputResultsToSCR

Reply:
no need to call. that method was wrongly called here and would have deleted the actual issueing of ESDT token from an smart contract. in case of outPutTransfer you need different operations here.

Reply:
ok